Changed FKs and related_names in Circuits app (#3090)
nautobot/nautobot: nautobot/docs/release-notes/version-2.0.md

#### Renamed Database Foreign Keys and Related Names ([#2520](https://github.com/nautobot/nautobot/issues/2520))

Some Foreign Key fields have been renamed to follow a more self-consistent pattern across the Nautobot app. This change is aimed to offer more clarity and predictability when it comes to related object database operations:

For example in v1.x to create a circuit object with `type` "circuit-type-1", you would do:

```python
Circuit.objects.create(
    cid="Circuit 1",
    provider="provider-1",
    type="circuit-type-1",
    status="active",
)
```

and to filter `Circuit` objects of `type` "circuit-type-2", you would do:

```python
Circuit.objects.filter(type="circuit-type-2")
```

Now in v2.x, we have renamed the Foreign Key field `type` on Circuit Model to `circuit_type`, because this naming convention made it clearer that this Foregin Key field is pointing to the model `CircuitType`. The same operations would look like:

```python
Circuit.objects.create(
    cid="Circuit 1",
    provider="provider-1",
    circuit_type="circuit-type-1",
    status="active",
)
```

```python
Circuit.objects.filter(circuit_type="circuit-type-2")
```

> Check out more Foreign Key related changes documented in this table [Renamed Database Fields](../installation/upgrading-from-nautobot-v1.md#renamed-database-fields)

Reviewer: ```suggestion
Check out more Foreign Key related changes documented in the table [Renamed Database Fields](../installation/upgrading-from-nautobot-v1.md#renamed-database-fields).
```

> Check out more Foregin Key related changes documented in this table [Renamed Foregin Key Fields](../installation/upgrading-from-nautobot-v1.md#renamed-foregin-key-fields)

Reviewer 2: ```suggestion
Check out more Foregin Key related changes documented in this table [Renamed Foreign Key Fields](../installation/upgrading-from-nautobot-v1.md#renamed-foreign-key-fields)
```

In addition to the changes made to Foreign Key fields' own names, some of their `related_names` are also renamed:

For example in v1.x, to query `Circuit` objects with `CircuitTermination` instances located in sites ["ams01", "ams02", "atl03"], you would do:

```python

> Circuit.objects.filter(terminations__site=["ams01", "ams02", "atl03"])

Reviewer 2: ```suggestion
Circuit.objects.filter(terminations__site__in=["ams01", "ams02", "atl03"])
```

```

Now in v2.x, we have renamed the Foreign Key field `circuit`'s `related_name` attribute `terminations` on `CircuitTermination` Model to `circuit_terminations`, the same operations would look like:

```python

> Circuit.objects.filter(circuit_terminations__site=["ams01", "ams02", "atl03"])

Reviewer 2: ```suggestion
Circuit.objects.filter(circuit_terminations__site__in=["ams01", "ams02", "atl03"])
```

```

> Check out more `related-name` changes documented in this table [Renamed Database Fields](../installation/upgrading-from-nautobot-v1.md#renamed-database-fields)

Reviewer: ```suggestion
Check out more `related-name` changes documented in the table [Renamed Database Fields](../installation/upgrading-from-nautobot-v1.md#renamed-database-fields).
```

> Check out more `related-name` changes documented in this table [Renamed Foregin Key Related Names](../installation/upgrading-from-nautobot-v1.md#renamed-foregin-key-related-names)

Reviewer 2: ```suggestion
Check out more `related-name` changes documented in this table [Renamed Database Fields](../installation/upgrading-from-nautobot-v1.md#renamed-database-fields)
```
